File: backend/src/main/java/ropold/backend/dto/DeviceDTO.java
package ropold.backend.dto;

import ropold.backend.model.DeviceStatus;
import ropold.backend.model.DeviceType;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public record DeviceDTO(
        UUID id,
        DeviceType type,
        String manufacturer,
        String modelName,
        String serialNumber,
        String inventoryNumber,
        LocalDate purchaseDate,
        DeviceStatus status,
        boolean defective,
        LocationDTO location,
        String notes,
        List<DeviceFileDTO> files
) {}

File: frontend/src/components/dto/DeviceDTO.ts
import { type DeviceType, type DeviceStatus } from "../models/DeviceModel.ts";
import { type LocationDTO } from "./LocationDTO.ts";
import { type DeviceFileDTO } from "./DeviceFileDTO.ts";

export type DeviceDTO = {
    id: string | null;
    type: DeviceType;
    manufacturer: string | null;
    modelName: string | null;
    serialNumber: string | null;
    inventoryNumber: string | null;
    purchaseDate: string | null;
    status: DeviceStatus;
    defective: boolean;
    location: LocationDTO | null;
    notes: string | null;
    files: DeviceFileDTO[] | null;
};